python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -206,16 +350,13 @@ def start_training(
                 is ``None`` then no checkpoint will be loaded.
             checkpoint_strategy (Optional[CheckpointStrategy]): The
                 configurations for saving checkpoints.
+            run_dir (Optional[Path]): The directory to use for this run.
+            checkpoint_id (Optional[int]): The checkpoint id to start with.
```

Comment:
"start with" is a little ambiguous - should this be the ID of the last successfully completed checkpoint?

Reply:
Not necessarily. It's just whatever checkpoint id you want to start with.

Reply:
Oh I think we might need to align on 0 vs. 1 indexing for checkpoint ID. I think with the current code writing to disk may break with the following path:

1. `checkpoint_id` = 0 (default)
2. `_process_checkpoint` calls `write_checkpoint` calls `latest_checkpoint_path` which will return `None` (which would then fail?).

This might be because we're using `latest_checkpoint_path`/`latest_checkpoint_id` for reading and writing. The previous logic was to use 0 as the magic number for "no checkpoint", and then it is incremented at the start of processing.

Reply:
Isn't `checkpoint_id` incremented before `write_checkpoint` is called?

Reply:
Yeah this would break if the `checkpoint_id` is -1. I can check for positive only.